db/R/utils.R

```diff
@@ -166,55 +166,131 @@ db.print.connections <- function() {
 ##' @return TRUE if database connection works; else FALSE
 ##' @export
 ##' @author David LeBauer, Rob Kooper
-db.exists <- function(params, write=TRUE) {
-	# open connection
-	con <- tryCatch({
-		invisible(db.open(params))
-	}, error = function(e) {
-		logger.error("Could not connect to database.\n\t", e)
-		invisible(NULL)
-	})
-	if (is.null(con)) {
-		return(invisible(FALSE))
-	}
-
-  if(FALSE){
-	# read a row from the database
-	read.result <- tryCatch({
-		invisible(db.query("SELECT * FROM users LIMIT 1", con))
-	}, error = function(e) {
-		logger.error("Could not query database.\n\t", e)
-		db.close(con)
-		invisible(NULL)
-	})
-	if (is.null(read.result)) {
-	  return(invisible(FALSE))
-	}
-	
-	# if requested write a row to the database
-	if (write) {
-		result <- tryCatch({
-			db.query(paste("UPDATE users SET created_at='", read.result$created_at, "' WHERE id=", read.result$id, sep=""), con)
-			invisible(TRUE)
-		}, error = function(e) {
-			logger.error("Could not write to database.\n\t", e)
-			invisible(FALSE)
-		})
-	} else {
-    result <- TRUE
+db.exists <- function(params, write=TRUE, table=NA) {
+  # open connection
+  con <- tryCatch({
+    invisible(db.open(params))
+  }, error = function(e) {
+    logger.error("Could not connect to database.\n\t", e)
+    invisible(NULL)
+  })
+  if (is.null(con)) {
+    return(invisible(FALSE))
   }
-
+  
+  #check table's privilege about read and write permission
+  user.permission <- tryCatch({
+    invisible(db.query(paste("select * from information_schema.role_table_grants where grantee='bety'"), con))
```

Comment:
@zhangwenx What it is about the grantee's role / permissions (`role_table_grants.permission_type`) that makes it useful here - are you looking for the `params$user`?

On ebi-forecast, `bety` is the only grantee (`select distinct grantee from information_schema.role_table_grants;`).